src/lib.rs

```diff
@@ -117,16 +117,30 @@ impl BarcodeScanner {
 		loop {
 			let events = self.device.fetch_events()
 				.map_err(|e| Error::new(format!("Failed to fetch events from input device: {e}")))?;
+
+			// Track the state of the shift keys and capslock
+			let mut left_shift_pressed = false;
+			let mut right_shift_pressed = false;
+			let mut capslock_on = false;
 			for event in events {
 				// Check if key is pressed (value 1 for the key pressed, velue 0 for the key released).
-				if event.event_type() == evdev::EventType::KEY && event.value() == 1 {
+				if event.event_type() == evdev::EventType::KEY {
 					// Create Key object based on the code.
 					let key_name = evdev::Key(event.code());
 
-					// Map key_name to the number or char.
-					if let Some(c) = key_to_str(key_name) {
-						self.buffer.push(c);
-					}
+					match key_name {
+						evdev::Key::KEY_LEFTSHIFT => left_shift_pressed = event.value() == 1,
+						evdev::Key::KEY_RIGHTSHIFT => right_shift_pressed = event.value() == 1,
+						evdev::Key::KEY_CAPSLOCK => capslock_on = event.value() == 1,
```

Comment:
It looks like capslock is reported as usual: press and release events match the actual key state. Not the "is capslock enabled" bit.

If I use a real keyboard, I do receive LED events. It may be some other components doing that: the Xorg server or the Wayland compositor. But that's fine. I think we should just follow the LED events and not look at the KEY_CAPSLOCK events.

Reply:
I would just say to drop it. Caps lock working differently to pressing shift would require more complexity in handling it anyway.

Also to my understanding the LED events come from whatever is handling the keyboard. The keyboard is not responsible for tracking the LED states, usually Xorg or a Wayland compositor.

Reply:
Right, sounds good to me. I highly doubt barcode scanners would use capslock anyway. And if they do they shouldn't >.<